kore/src/Kore/Log/ErrorRewriteRuleId.hs

```diff
@@ -0,0 +1,42 @@
+{- |
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+
+-}
+
+module Kore.Log.ErrorRewriteRuleId
+    ( ErrorRewriteRuleId
+    , errorRewriteRuleId
+    ) where
+
+import Prelude.Kore
+
+import Data.Text.Prettyprint.Doc as Pretty
+
+import Kore.Step.RulePattern
+    ( RewriteRule
+    , getRewriteRule
+    )
+import Kore.Syntax.Variable
+    ( Variable
+    )
+
+import Log
+
+newtype ErrorRewriteRuleId =
+    ErrorRewriteRuleId { getRule :: RewriteRule Variable }
+
+instance Pretty ErrorRewriteRuleId where
+    pretty errorRewriteRule =
+        Pretty.vsep
+            [ "Found semantic rule with the same left- and right-hand side:"
+            , Pretty.pretty . getRewriteRule . getRule $ errorRewriteRule
```

Comment:
Let's print the original source location here, instead of the full rule. We should also add another line of explanation, such as:
```suggestion
            , Pretty.pretty . getRewriteRule . getRule $ errorRewriteRule
            , "Execution would not terminate when the rule applies."
```